feat(hl): add 'TreesitterContextLineNumberBottom'

Danielkonge opened the pull request:
This closes #383.

For now I replaced `highlight_bottom` with `highlight_lno_bottom`, which will change the current behavior a bit, but also allow for more precise configuration by the user.

If you prefer to keep `highlight_bottom` as it was, I can add it back in together with the new function.

lewis6991 commented on lua/treesitter-context/render.lua:
```diff
@@ -232,6 +232,16 @@ local function highlight_bottom(bufnr, row)
 
+---@param bufnr integer
+---@param row integer
+local function highlight_lno_bottom(bufnr, row)
```
This is a bit repetitive. Can you combine it with the function above?

Danielkonge replied:
I can add a third variable to the function above (e.g. a boolean telling me which hl_group to choose), if that is better?

lewis6991 replied:
I would say add a string

lewis6991 commented on lua/treesitter-context.lua:
```diff
@@ -188,6 +188,7 @@ local function init()
 
   api.nvim_set_hl(0, 'TreesitterContext', { link = 'NormalFloat', default = true })
   api.nvim_set_hl(0, 'TreesitterContextLineNumber', { link = 'LineNr', default = true })
+  api.nvim_set_hl(0, 'TreesitterContextLineNumberBottom', { link = 'NONE', default = true })
```
Maybe link to `TreesitterContextBottom` by default?

Danielkonge commented:
I tried addressing the comments now. Please just let me know if you want to change anything else. (Also, since it was such a small PR I just squashed the commits immediately.)

lewis6991 commented:
Thanks
